util/copyfiles.go

```diff
@@ -73,7 +76,25 @@ func CopyFile(to, src string) error {
 		}
 	}()
 
-	// Write source contents to destination
-	_, err = io.Copy(w, r)
-	return err
+	if 0 < len(redactions) {
+		scanner := bufio.NewScanner(r)
+		// Scan, redact, and write each line of the src file
+		for scanner.Scan() {
+			bts := scanner.Bytes()
+			bts = append(bts, '\n')
```

Comment:
This tripped me up until I tested it myself. Didn't know that scanner.Scan() chews up newlines.

Reply:
Yeah this was unintuitive - I had to break a file first to see it. The scanner impl. takes or applies a default split function  and like many split operations the delimiter is not in the result. I bet it's possible to work around this by passing our own split function.